infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -200,7 +201,12 @@ public SequencerServer(ServerContext serverContext,
 
     @Override
     protected void processRequest(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
-        executor.submit(() -> getHandlerMethods().handle(req, ctx, r));
+        // Capture total cost: queueing delay + handler cost
+        long start = System.nanoTime();
+        executor.submit(() -> {
+            getHandlerMethods().handle(req, ctx, r);
+            MicroMeterUtils.time(Duration.ofNanos(System.nanoTime() - start), "sequencer.loop.latency");
```

Comment:
wouldnt it get measured in -
https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/RequestHandlerMethods.java#L163

Reply:
No. That just measures the time it takes to execute the handler only, but my patch measures the time it takes to execute the handler and queue delay.

The way it works as follows:
1. Netty IO thread reads a message
2. Dispatch message to the sequencer work queue
3. Sequencer threads picks up work items from the queue and processes them in FIFO order

Notice that the start time stamp is taken before the item is submitted to the queue and the end time stamp is after the handler is executed, which captures the **total** latency for servicing a sequencer request.